### PR TITLE
JDK Home

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Comment by henryju** [comment]
@m-g-sonar feel free to review as well ;)

File: src/main/java/org/sonarqube/gradle/SonarQubeExtension.java [hunk 1]
```diff
@@ -20,6 +20,9 @@
 package org.sonarqube.gradle;
 
 import org.gradle.api.Action;
+import org.gradle.api.provider.Property;
```
Comment (mark-rekveld-sonarsource) [inline src/main/java/org/sonarqube/gradle/SonarQubeExtension.java:23]: are these imports needed?

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java [hunk 2]
```diff
@@ -27,11 +27,13 @@
 import org.gradle.api.internal.ConventionMapping;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
+import org.gradle.api.model.ObjectFactory;
```
Comment (mark-rekveld-sonarsource) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java:30]: are these imports needed?

File: src/main/java/org/sonarqube/gradle/ToolchainUtils.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    return Optional.empty();
+  }
+
+  public static boolean hasToolchains(JavaCompile compileTask) {
+    return compileTask.getJavaCompiler().isPresent();
+  }
+
+  // Inspired by
+  // https://github.com/gradle/gradle/blob/d3e4faca3df507176b67d9b3bb3ee91bf2aa070c/subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java#L400
```
Comment (mark-rekveld-sonarsource) [inline src/main/java/org/sonarqube/gradle/ToolchainUtils.java:55]: good thing you found this change!

**Review by mark-rekveld-sonarsource — APPROVED** [review]
Looks goo to me.

I'm not fluent in Spock yet, although the tests look good to me I would like another reviewer with more Spock experience to look at the Spock tests.

File: src/main/java/org/sonarqube/gradle/SonarQubeExtension.java (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarqube/gradle/SonarQubeExtension.java]: Good catch, remains from previous experiments.

**Review by malena-ebert-sonarsource — APPROVED** [review]

**Pull request merged**